### ISSUE TITLE
Problem with selecting fully qualified column names

### ISSUE THREAD

**Issue by theme1256** [dolthub/dolt#7247]
It looks like it's not possible, at least in Java, to use full names for selected fields, when using the mysql-driver.

Sample Data:
table1:
| ID | Title |
|--------|--------|
| 1 | Sample Title |

table2:
| ID | Text |
|--------|--------|
| 1 | Sample text 1 |
| 1 | Sample text 2|

If the database is MySQL, the following code works, but not, when the database is Dolt, it throws S0022 "Column 'table1.ID' not found.".

```java
String sql = "SELECT * FROM table2 JOIN table1 ON table1.ID = table2.ID";
try (ResultSet rs = conn.createStatement().executeQuery(sql)) {
    int id = rs.getInt("table1.ID");
} catch (SQLException ex) { ex.print }
```

If I re-write the query to `SELECT table1.ID, Title, Text FROM table2 JOIN table1 ON table1.ID = table2.ID` It still doesn't work  
If I have to re-write line 3 to `rs.getInt("ID")` for it to work.

**Comment by timsehn** [comment]
We will fix this today. Keep the bugs coming.

--Tim

**Comment by bpf120** [comment]
@theme1256 , Thanks for all the bugs! We'd love to learn about your Dolt use case too. Feel free to share here, [email me](mailto:brianf@dolthub.com), or swing by our [Discord](https://discord.gg/gqr7K4VNKe) if you want to share.

**Comment by fulghum** [comment]
My first guess is that MySQL is sending back additional metadata about the table names that Dolt is not. I'm going to set up a repro and look at the wire logs between MySQL and Dolt and will update with what I find.

**Comment by fulghum** [comment]
I think I've got this figured out. As expected, we were missing some field metadata in the wire response that was enabling the Java MySQL Connector library to understand which `ID` field was coming from the `table1` table. I've got a repro with MySQL and Dolt, and I've got some changes in my workspace that get the repro working correctly with Dolt. 

I'm going to dig into our tests next and make sure we have good coverage to prevent a regression here. In particular, we have existing tests for the MySQL Connector libraries in various languages, and I'll get those updated to cover using qualified column name references. 

I'm expecting we'll be able to get this fix released later today for you. Thank you for taking the time to report this so we could dig in and fix it! 🙏
